### PR TITLE
exit: fix race zombie reaper

### PR DESCRIPTION
**Pull request by giuseppe**
fix a race condition where all the processes spawned by the OCI
runtime are not terminated yet and reap_children() would not clean
them up.

If an --exit-delay is specified, these proceses would be left
uncleaned until the delay expires.

When used for an exec session, the zombie process would prevent the
container namespaces to be cleaned up.

Closes: https://github.com/containers/conmon/issues/290

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM, great work!

File: src/ctr_exit.c [hunk 1]
```diff
@@ -156,9 +150,18 @@ void do_exit_command()
 	if (exit_pid) {
 		int ret, exit_status = 0;
 
-		do
-			ret = waitpid(exit_pid, &exit_status, 0);
-		while (ret < 0 && errno == EINTR);
+		/* even though we've already registered reap_children,
+		 * atexit() runs functions in reverse, so we need to
+		 * manually call here. Repeated calls will not cause trouble
+		 */
```
Comment (haircommander) [inline src/ctr_exit.c:156]: I think this comment needs updating because we're not calling reap_children here

**Comment by rhatdan** [comment]
LGTM

**Comment by lsm5** [comment]
@giuseppe @haircommander mind cutting a new version with this? /cc @edsantiago

**Comment by edsantiago** [comment]
I'll second this. We are starting to test podman-remote in gating, and gating tests are guaranteed to fail with old conmon.

**Comment by haircommander** [comment]
opened https://github.com/containers/conmon/pull/296

**Comment by lsm5** [comment]
@haircommander could you push the new tag as well?

Also, I see the dependabot security alert for an older buildah in there? Is that something worth handling before we build the new tag?

**Pull request merged**